src/JuMP.jl

```diff
@@ -738,5 +729,19 @@ include("print.jl")
 # Nonlinear-specific code
 include("nlp.jl")
 
+
+typealias _getTypes Union{NonlinearExpression,NonlinearParameter, AffExpr}
+getValue{T<:_getTypes}(arr::Array{T}) = map(getValue, arr)
+
+typealias _setTypes Union{NonlinearParameter, Variable}
+function setValue{T<:_setTypes}(set::Array{T}, val::Array)
```

Comment:
Similar to above, I think I would prefer using ``T<:AbstractJuMPScalar`` (with ``NonlinearParameter <: AbstractJuMPScalar`` elsewhere), rather than this type union.